pkgs/development/compilers/codon/default.nix

```diff
@@ -0,0 +1,151 @@
+{ cacert
+, clang_15
+, cmake
+, fetchFromGitHub
+, git
+, lib
+, lld_15
+, ninja
+, nix-update-script
+, perl
+, python3
+, stdenv
+}:
+
+let
+  version = "0.15.5";
+
+  src = fetchFromGitHub {
+    owner = "exaloop";
+    repo = "codon";
+    rev = "v${version}";
+    sha256 = "sha256-/IUGX5iSRvZzwyRdkGe0IVHp44D+GXZtbkdtswekwSU=";
+  };
+
+  depsDir = "deps";
+
+  codon-llvm = stdenv.mkDerivation
+    {
+      pname = "codon-llvm";
+      version = "unstable-2022-09-23";
+
+      src = fetchFromGitHub {
+        owner = "exaloop";
+        repo = "llvm-project";
+        rev = "55b0b8fa1c9f9082b535628fc9fa6313280c0b9a";
+        sha256 = "sha256-03SPQgNdrpR6/JZ5aR/ntoh/FnZvCjT/6bTAcZaFafw=";
+      };
+
+      nativeBuildInputs = [
+        clang_15
+        cmake
+        git
+        lld_15
+        ninja
+        python3
+      ];
+
+      preConfigure = ''
+        cmakeFlagsArray+=(
+          "-DCMAKE_CXX_COMPILER=clang++"
+          "-DCMAKE_C_COMPILER=clang"
+          "-DLLVM_ENABLE_RTTI=ON"
+          "-DLLVM_ENABLE_TERMINFO=OFF"
+          "-DLLVM_ENABLE_ZLIB=OFF"
+          "-DLLVM_INCLUDE_TESTS=OFF"
+          "-DLLVM_TARGETS_TO_BUILD=all"
+          "-DLLVM_USE_LINKER=lld"
+          "-S ../llvm"
+        )
+      '';
+
+      cmakeFlags = [
+      ];
```

Comment:
```suggestion
      cmakeFlags = [
        "-DCMAKE_CXX_COMPILER=clang++"
        "-DCMAKE_C_COMPILER=clang"
        "-DLLVM_ENABLE_RTTI=ON"
        "-DLLVM_ENABLE_TERMINFO=OFF"
        "-DLLVM_ENABLE_ZLIB=OFF"
        "-DLLVM_INCLUDE_TESTS=OFF"
        "-DLLVM_TARGETS_TO_BUILD=all"
        "-DLLVM_USE_LINKER=lld"
        "-S ../llvm"
      ];
```

Reply:
I used them at first, but the [`nixpkgs-hammer`](https://github.com/jtojnar/nixpkgs-hammering) tool suggested to use `cmakeFlagsArray` because of the spaces. [Here](https://github.com/jtojnar/nixpkgs-hammering/blob/243b81c687aac33d6716957c0cd2235c81631044/overlays/no-flags-spaces.nix)'s the check.

Reply:
Shouldn't the space be resplit anyway?

Reply:
Eh, I guess. Anyway, changed to `cmakeFlags` since it doesn't matter.